### ISSUE TITLE
M2 force-calling multi-allelic sites using GGA mode results in erroneous mappinq_quality filter if one allele does not have any reads mapped

### ISSUE THREAD

**Issue by fpbarthel** [broadinstitute/gatk#5695]
### Affected tool(s) or class(es)
`Mutect2` and `FilterMutectCalls`

### Affected version(s)
GATK 4.1.0.0 release (though I suspect this was also in 4.0.x)

### Description 
M2 force-calling multi-allelic sites using GGA mode results in automatic `mappinq_quality` filter if one allele does not have any reads mapped, even if the other variant allele has good mapping quality.

In my testing this happened when I force-called the following IDH1 mutations:

```
#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO
2       209113112       .       C       A       .       .       .
2       209113112       .       C       T       .       .       .
```

The first listed is the common IDH1 R132H variant and the second listed is the less common but equally pathogenic R132L variant. The purpose for force-calling both variants is that one wants to absolutely exclude the possibility of either variant at this site.

In most my tests, variants would fail `FilterMutectCalls` because only one of the multi-allelic alleles would be present, eg.

*(representative except of `Mutect2` + `FilterMutectCalls` output)*

```
VAR  	2	209113112	.	C	A,T
FILTER	mapping_quality;read_position
INFO	MMQ=60,0,60
FORMAT	GT:AD   0/1/2:56,0,47
```

Only in rare cases would the variant site `PASS`, and that's when there would be at least one variant read in the second allele, eg:

```
VAR  	2	209113112	.	C	A,T
FILTER	PASS
INFO	MMQ=60,60,60
FORMAT	GT:AD   0/1/2:28,1,16
```

#### Steps to reproduce
To reproduce this specific example you will need a BAM file with a C>T mutation at `2:209113112`, and no reads with an `A` at this position.

You will then need to run M2 and `FilterMutectCalls` with `--genotyping-mode GENOTYPE_GIVEN_ALLELES --alleles ...` set to a VCF with both lines as shown in the description.

#### Expected behavior
`FilterMutectCalls` should not apply a `mapping_quality` filter if one of the alleles would pass. Perhaps the mapping quality filter could ignore alleles with zero mapped reads, rather than assigning a MMQ of zero by default? *Note that this issue would also not happen if this multi-allelic site was split, since in that case only the C>A variant would get filtered but not the C>T.*

#### Actual behavior
See description

**Comment by davidbenjamin** [comment]
@fpbarthel Good catch!  This is definitely the wrong behavior.

**Comment by davidbenjamin** [comment]
@fpbarthel This will be fixed once PR #5743 goes in.  Thanks for spotting it!

**Comment by fpbarthel** [comment]
Thanks!
